homeassistant/components/ping/binary_sensor.py

```diff
@@ -142,7 +149,11 @@ def ping(self):
             match = PING_MATCHER.search(str(out).split("\n")[-1])
             rtt_min, rtt_avg, rtt_max, rtt_mdev = match.groups()
             return {"min": rtt_min, "avg": rtt_avg, "max": rtt_max, "mdev": rtt_mdev}
-        except (subprocess.CalledProcessError, AttributeError):
+        except (
+            subprocess.TimeoutExpired,
+            subprocess.CalledProcessError,
```

Comment:
On timeout, we need kill the process and wait again to prevent for zombi

Reply:
Will make a util for this

Reply:
done